### ISSUE TITLE
#include <boost/range/mfc.hpp> causes a compilation error

### ISSUE THREAD

**Issue by gabrielbocek** [boostorg/range#143]
Hi,
I updated the visual C++ compiler from 17.0.6  to 17.4.6 and I'm getting a compilation error by including <boost/range/mfc.hpp>:

1>\boost\range\detail\microsoft.hpp(626,33): error C2440: 'return': cannot convert from 'const void *' to 'const void *&'
1>\boost\range\detail\microsoft.hpp(624,1): message : while compiling class template member function 'const void *&boost::range_detail_microsoft::list_iterator<const X,const void *,boost::use_default,boost::use_default>::dereference(void) const'
1>        with
1>        [
1>            X=CPtrList
1>        ]
1>\boost\iterator\iterator_facade.hpp(631,11): message : see reference to function template instantiation 'const void *&boost::range_detail_microsoft::list_iterator<const X,const void *,boost::use_default,boost::use_default>::dereference(void) const' being compiled
1>        with
1>        [
1>            X=CPtrList
1>        ]
1>\boost\range\mfc.hpp(881,1): message : see reference to class template instantiation 'boost::range_detail_microsoft::list_iterator<const X,const void *,boost::use_default,boost::use_default>' being compiled
1>        with
1>        [
1>            X=CPtrList
1>        ]

The project is compiled with /std:c++20 switch.

This is easily reproducible - it is just needed to create a C++ console application, add the include to <boost/range/mfc.hpp> to one cpp file, add boost directory to include directories and set the compiler flag /std:c++20.

Best regards,
Gabriel

**Comment by tobias-loew** [comment]
I made a non-intrusive fix #144. But it isn't merged yet. 🙁

**Comment by gabrielbocek** [comment]
@tobias-loew  How can we speed up the merge? I see that you made the fix in April '23.

**Comment by tobias-loew** [comment]
It just happened! (Even without the need for defining the macro: the old code was broken anyway.)

**Comment by gabrielbocek** [comment]
Somehow this fix is still not part of the Boost 1.84. Can you please clarify?

**Comment by tobias-loew** [comment]
@neilgroves Any chance to get #144 fix into the boost-release? 
(There are several other commits also waiting for release. The last merge into master was May 2021)

**Comment by neilgroves** [comment]
Yes I can merge to master when I return to the UK in a few days.

Thanks,
Neil
